packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java

```diff
@@ -622,9 +623,10 @@ public void setStateWrapper(StateWrapper stateWrapper) {
   public void draw(Canvas canvas) {
     if (mEndFillColor != Color.TRANSPARENT) {
       final View contentView = getContentView();
-      if (mEndBackground != null && contentView != null && contentView.getBottom() < getHeight()) {
-        mEndBackground.setBounds(0, contentView.getBottom(), getWidth(), getHeight());
-        mEndBackground.draw(canvas);
+      if (contentView != null && contentView.getBottom() < getHeight()) {
+        Paint paint = new Paint();
+        paint.setColor(mEndFillColor);
+        canvas.drawRect(0, contentView.getBottom(), getWidth(), getHeight(), paint);
```

Comment:
Why do we do this if we are creating `mEndBackground` from new color?

Reply:
That was an oversight. I meant to remove `mEndBackground` as we need to draw on the canvas directly order to use paint.